packages/react-instantsearch-hooks/src/lib/useIndex.ts

```diff
@@ -37,3 +43,39 @@ export function useIndex(props: UseIndexProps) {
 
   return indexWidget;
 }
+
+function useParentIndex({ parentIndexId }: UseIndexOwnProps) {
+  const physicalParentIndex = useIndexContext();
+  const search = useInstantSearchContext();
+
+  if (parentIndexId) {
```

Comment:
if we want to support null == mainIndex:

```suggestion
  if (parentIndexId === null) {
    return searchMainIndex;
  }
  if (parentIndexId) {
```